lib/puppet/provider/base_dsc/powershell.rb

```diff
@@ -52,13 +52,22 @@ def create
     script_content = ps_script_content('set')
     Puppet.debug "\n" + script_content
     output = powershell(powershell_args, script_content)
-    Puppet.debug "Dsc Resource Return: #{output}"
+    if ['true','false'].include?(output.to_s.strip.downcase)
+      if output.to_s.strip.downcase == 'true'
+        # reboot is required
```

Comment:
If a DSC Resource indicates the system needs a reboot, we should tell puppet to do so in both create and destroy. How do we do this?